File: src/main/java/org/blacksun/graph/channel/Channel.java
package org.blacksun.graph.channel;

import org.blacksun.graph.node.GraphNode;
import org.blacksun.utils.StringRepresentable;

/**
 * Represents connection between two nodes from one network.
 * <br>
 * Parameters of channel are mutable but connected nodes could not be changed.
 */
public interface Channel extends StringRepresentable {
    double ERRORS_AMOUNT = 0.1;

    GraphNode getFromNode();

    GraphNode getToNode();

    void setWeight(int weight);

    int getWeight();

    void setErrors(double errors);

    double getErrors();

    void setUsed(boolean used);

    boolean isUsed();

    void connect();

    void remove();

    @Override
    default String stringRepresentation() {
        return toString();
    }
}

File: src/main/java/org/blacksun/graph/channel/Channel.kt
package org.blacksun.graph.channel

import org.blacksun.graph.node.GraphNode
import org.blacksun.utils.Selectable
import org.blacksun.utils.StringRepresentable

/**
 * Represents connection between two nodes from one network.
 * <br></br>
 * Parameters of channel are mutable but connected nodes could not be changed.
 */
interface Channel : StringRepresentable, Selectable {

    val fromNode: GraphNode

    val toNode: GraphNode

    var weight: Int

    var errors: Double

    var isUsed: Boolean

    fun connect()

    fun remove()

    override fun stringRepresentation(): String {
        return toString()
    }

    companion object {
        const val ERRORS_AMOUNT = 0.1
    }
}
